bin/sozo/src/ops/events.rs

```diff
@@ -40,124 +39,437 @@ pub async fn execute(
 
     if let Some(events_map) = events_map {
         parse_and_print_events(res, events_map)?;
-    } else {
-        println!("{}", serde_json::to_string_pretty(&res)?);
     }
 
     Ok(())
 }
 
 fn parse_and_print_events(
     res: starknet::core::types::EventsPage,
-    events_map: HashMap<String, Vec<Event>>,
+    events_map: HashMap<String, Vec<Token>>,
 ) -> Result<()> {
     println!("Continuation token: {:?}", res.continuation_token);
     println!("----------------------------------------------");
     for event in res.events {
         if let Some(e) = parse_event(event.clone(), &events_map) {
             println!("{e}");
-        } else {
-            // Couldn't parse event
-            println!("{}", serde_json::to_string_pretty(&event)?);
         }
     }
     Ok(())
 }
 
+fn parse_core_basic(
+    cb: &CoreBasic,
+    value: &FieldElement,
+    include_felt_string: bool,
+) -> Result<String, String> {
+    match cb.type_name().as_str() {
+        "felt252" => {
+            let hex = format!("{:#x}", value);
+            match parse_cairo_short_string(value) {
+                Ok(parsed) if !parsed.is_empty() && include_felt_string => {
+                    Ok(format!("{} \"{}\"", parsed, hex))
+                }
+                _ => Ok(format!("\"{}\"", hex)),
+            }
+        }
+        "bool" => {
+            if *value == FieldElement::ZERO {
+                Ok("false".to_string())
+            } else {
+                Ok("true".to_string())
+            }
+        }
+        "ClassHash" | "ContractAddress" => Ok(format!("{:#x}", value)),
+        "u8" | "u16" | "u32" | "u64" | "u128" | "usize" | "i8" | "i16" | "i32" | "i64" | "i128" => {
+            Ok(value.to_string())
+        }
+        _ => Err(format!("Unsupported CoreBasic type: {}", cb.type_name())),
```

Comment:
```suggestion
        _ => Err(anyhow!("Unsupported CoreBasic type: {}", cb.type_name())),
```
lets use the `anyhow!` macro when returning an error